app/models/bulkrax/oai_entry.rb

```diff
@@ -44,17 +38,30 @@ def build_metadata
       end
       add_metadata('thumbnail_url', thumbnail_url)
 
+      parsed_metadata[Bulkrax.system_identifier_field] ||= [record.header.identifier]
+      parsed_metadata['contributing_institution'] = [contributing_institution]
+
       add_visibility
       add_rights_statement
-      self.parsed_metadata['source'] ||= [record.header.identifier]
-      self.parsed_metadata['contributing_institution'] = [contributing_institution]
+      add_collections
 
-      if collection.present?
-        self.parsed_metadata['collections'] ||= []
-        self.parsed_metadata['collections'] << {id: self.collection.id}
-      end
+      parsed_metadata
+    end
 
-      return self.parsed_metadata
+    # Retrieve list of collections for the entry; add to collection_ids
+    # If OAI-PMH doesn't return setSpec in the headers for GetRecord, use parser.collection_name
+    #   in this case, if 'All' is selected, records will not be added to a collection.
```

Comment:
does that make the most sense or would it make more sense if all is selected for it to act the same way sets.blank? acts (ie add to default collection?)

Reply:
So I think in my refactor, we don't have a default collection. This is one I wanted to chat to you about.

Reply:
I see, we agree that there is not a default possible in the 'all' case where no set of collections is chosen on an individual entry.